crypto/evp/s_lib.c

```diff
@@ -25,18 +26,16 @@ int EVP_SKEY_export(const EVP_SKEY *skey, int selection,
         return 0;
     }
 
-    if (skey->skeymgmt == NULL) {
-        ERR_raise(ERR_LIB_EVP, ERR_R_PASSED_INVALID_ARGUMENT);
-        return 0;
-    }
-
     return evp_skeymgmt_export(skey->skeymgmt, skey->keydata, selection, export_cb, export_cbarg);
 }
 
-EVP_SKEY *evp_skey_alloc(void)
+static EVP_SKEY *evp_skey_alloc(EVP_SKEYMGMT *skeymgmt)
 {
     EVP_SKEY *skey = OPENSSL_zalloc(sizeof(EVP_SKEY));
 
+    if (skey == NULL || !ossl_assert(skeymgmt != NULL))
```

Comment:
Won't this leak `skey` if `skeymgmt` is passed in as `NULL`?

Reply:
Fixed